geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateDefinedIndexesCommand.java

```diff
@@ -80,12 +80,25 @@ public void updateClusterConfig(String group, CacheConfig config, Object configO
     }
 
     for (RegionConfig.Index index : updatedIndexes) {
-      RegionConfig region = config.findRegionConfiguration(index.getFromClause());
+      String regionPath = getValidRegionName(index.getFromClause(), config);
+      RegionConfig region = config.findRegionConfiguration(regionPath);
       if (region == null) {
         throw new IllegalStateException("RegionConfig is null");
       }
 
       region.getIndexes().add(index);
     }
   }
+
+  String getValidRegionName(String regionPath, CacheConfig cacheConfig) {
+    // Check to see if the region path contains an alias e.g "/region1 r1"
+    // Then the first string will be the regionPath
+    String[] regionPathTokens = regionPath.trim().split(" ");
+    regionPath = regionPathTokens[0];
+    // check to see if the region path is in the form of "--region=region.entrySet() z"
+    while (regionPath.contains(".") && cacheConfig.findRegionConfiguration(regionPath) == null) {
```

Comment:
Since you're already getting the RegionConfig in this method, can you just return it instead of the regionPath? That'll avoid having to get it again in the caller.